File: Python/1750.py
class Solution:
    def minimumLength(self, s: str) -> int:
        left = 0
        right = len(s)-1

        while left < right and s[left] == s[right]:
            cur = s[left]
            while left <= right and cur == s[left]:
                left += 1
            while left <= right and cur == s[right]:
                right -= 1

        return right-left+1

# Topic: Two pointers
# Time complexity: O(n)
# Space complexity: O(1)

File: C++/1750.cpp
#include <string>

class Solution {
public:
    int minimumLength(std::string s) {
        int left = 0, right = s.size()-1;

        while(left < right && s[left] == s[right]){
            char cur = s[left];
            while(left <= right && cur == s[++left]){}
            while(left <= right && cur == s[--right]){}
        }
        return right-left+1;
    }
};

// Topic: Two Pointers
// Time Complexity: O(n)
// Space Complexity: O(1)